### PR TITLE
[SMALLFIX] Reorder common and client in alluxio-core-server

### PR DESCRIPTION
**Pull request by peisun1115**

### CODE REVIEW

**Comment by peisun1115** [comment]
@jsimsa @apc999

File: core/server/pom.xml [hunk 1]
```diff
@@ -35,14 +35,16 @@
       <groupId>com.google.protobuf</groupId>
       <artifactId>protobuf-java</artifactId>
     </dependency>
+    <!-- alluxio-core-common should come before alluxio-core-client as alluxio-core-client shade some dependencies
```
Comment (jsimsa) [inline core/server/pom.xml]: please also add a TODO to remove the alluxio-core-client dependency as having core-common be in the dependency tree twice -- once shaded through alluxio-core-client and once unshaded directly -- is error prone

Comment (jsimsa) [inline core/server/pom.xml]: otherwise, LGTM

Comment (peisun1115) [inline core/server/pom.xml]: done. thanks

**Pull request merged**